### PR TITLE
Added uuid to User object, added Report object

### PR DESCRIPTION
**Pull request by darbrius**

### CODE REVIEW

File: server/controllers/DataAccessLayer.js [hunk 1]
```diff
@@ -56,12 +57,12 @@ exports.AddUserToFile = function(user) {
...
 exports.UpdateUser = function(user) {
     // Find the index in the cache for the original User object
-    let index = cachedUsers.findIndex(x => x.username === user.username && x.email === user.email);
+    let index = cachedUsers.findIndex(x => x.id === user.id);
     // Get a pointer to the original User object
     let originalUser = cachedUsers[index];
 
     // Create a string that will be searched for within the text file
-    let originalUserString = originalUser.username + ";" + originalUser.password + ";" + 
+    let originalUserString = originalUser.id + ";" + originalUser.username + ";" + originalUser.password + ";" +
```
Comment (chankk) [inline server/controllers/DataAccessLayer.js]: I found these semi-colons a little jarring to look at, would commas be a better fit (similar to a CSV)

File: server/model/User.js [hunk 2]
```diff
@@ -1,6 +1,10 @@
+const { v1: uuid } = require('uuid');
+
 class User {
```
Comment (chankk) [inline server/model/User.js:3]: Are you missing the number of losses and totals games played? If hands only includes wins, there might be a better variable name.

**Review by chankk — CHANGES_REQUESTED** [review]
Potentially missing data on Users.

File: server/model/User.js (hunk 2 above)
Comment (darbrius) [reply server/model/User.js:3]: My bad, forgot to  pull master before I worked on this. Fixed

File: server/controllers/DataAccessLayer.js (hunk 1 above)
Comment (darbrius) [reply server/controllers/DataAccessLayer.js]: I've converted them to commas

File: server/controllers/DataAccessLayer.js [hunk 3]
```diff
@@ -18,19 +18,20 @@ exports.ReadUsersFile = function() {
         // Convert the buffer recieved to an ascii string
         let lineString = line.toString('ascii');
         // Split the string by a semicolon
```
Comment (chankk) [inline server/controllers/DataAccessLayer.js]: `// Split the string by a comma`

File: server/data/Users.txt [hunk 4]
```diff
@@ -1 +1 @@
-Jim;hunter123;Tim@tim.com;212;4;10;2020-03-17T14:15:41.727Z;2020-03-17T00:00:00.000Z;
\ No newline at end of file
+2eda5ab0-6a63-11ea-a4b4-ebd4268cc60a,Jim,hunter123,Tim@tim.com,212,4,10,2020-03-17T14:15:41.727Z,2020-03-17T00:00:00.000Z,
```
Comment (chankk) [inline server/data/Users.txt]: should this have been updated?

**Review by chankk — APPROVED** [review]

**Pull request merged**